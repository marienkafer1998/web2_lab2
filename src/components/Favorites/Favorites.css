.adder {
    width: 300px;
    font-size: 20px;
    padding: 6px 0 4px 10px;
    border: 1px solid #cecece;
    background: #F6F6f6;
    border-radius: 8px;
    margin-right: 10px;
}

.fav {
    color: #74878c;
    font-weight: 500;
    font-size: 3em;
}