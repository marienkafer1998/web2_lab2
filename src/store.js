import { createStore, combineReducers, applyMiddleware } from "redux";
import logger from "redux-logger";
import thunk from "redux-thunk";
import { LOCAL_STORAGE_KEY } from "./LocalStorage";

import geoReducer from "./redux/geoReducer";
import favReducer from "./redux/favReducer";

const store = createStore(
    combineReducers({geo: geoReducer, fav: favReducer}),
    applyMiddleware(logger, thunk)
);



store.subscribe(() => {
    localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify([...store.getState().fav.favorites.keys()]));
});

export default store;
