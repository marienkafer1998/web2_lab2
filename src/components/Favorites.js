import React from "react";
import { connect } from "react-redux";

import AddFavorite from "./AddFavorite";
import Weather from "./Weather";
import { addFavorite, deleteFavorite, loadWeatherWithName } from "../redux/actions/actions";



class Favorites extends React.Component {

    render() {
        return (
            <div>
                <h1>Favorites</h1>
                <AddFavorite onSubmit={(e) => this.handleAddFavorite(e)} />

                <div>
                    {
                        [...this.props.favorites.entries()].map((entry) => {
                            return (
                                <Weather
                                    onFetch={() => this.props.loadWeatherWithName(entry[0])}
                                    onDelete={() => this.props.deleteFavorite(entry[0])}
                                    weather={entry[1]} />
                            );
                        })
                    }
                </div>
            </div>
        );
    }

    handleAddFavorite(e) {
        e.preventDefault();
        const cityName = e.currentTarget.elements.cityName.value;
        this.props.addFavorite(cityName);
        console.log(this.props.favorites.entries());

    }
}


function mapStateToProps(state) {
    return {
        favorites: state.fav.favorites,
        error: state.fav.error
    };
}

function mapDispatchToProps(dispatch) {
    return {
        addFavorite: (cityName) => {
            dispatch(addFavorite(cityName));
        },

        deleteFavorite: (cityName) => {
            dispatch(deleteFavorite(cityName));
        },

        loadWeatherWithName: (cityName) => {
            dispatch(loadWeatherWithName(cityName));
        }
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Favorites);