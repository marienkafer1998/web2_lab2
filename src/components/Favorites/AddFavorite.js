import React from "react";
import "./Favorites.css";
import Button from "reactstrap/es/Button";



export default function AddFavorite(props) {
    return (
        <form  onSubmit={(e) => props.onSubmit(e)}>
            <input  text-transform="capitalize" className={"adder"} type="text" name="cityName" placeholder="Add new city" required={true} />
            <Button type="submit" >+</Button>
        </form>
    );
}

