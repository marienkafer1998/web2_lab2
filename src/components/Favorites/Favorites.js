import React from "react";
import { connect } from "react-redux";
import {
    Container, Row, Col
} from 'reactstrap';
import AddFavorite from "../AddFavorite/AddFavorite";
import Weather from "../Weather/Weather";
import {addFavorite, deleteFavorite, loadFavourites, loadWeatherWithName} from "../../redux/actions";
import "../Weather/Weather.css";



class Favorites extends React.Component {
    componentDidMount() {
        this.props.loadFavourites();
    }

    render() {
        console.log(this.props.favorites.entries());
        let header, body;
        header = <Container>
            <Row>
                <Col>
                <h1 className={"fav"}>Favorites</h1>
                </Col>
                <Col sm={{ size: 5, offset: 1 }}>
                    <AddFavorite onSubmit={(e) => this.handleAddFavorite(e)} />
                </Col>
            </Row>
        </Container>;
        body =
            <Row>
            {
                [...this.props.favorites.entries()].map((entry) => {
                    return (
                        <Col xs="6">
                        <Weather
                            onFetch={() => this.props.loadWeatherWithName(entry[0])}
                            onDelete={() => this.props.deleteFavorite(entry[0])}
                            weather={entry[1]} /> </Col>
                    );
                })
            }
</Row>



        return (
            <Container className="favorite-weather">
                {header}
                {body}
            </Container>


        );
    }

    handleAddFavorite(e) {
        e.preventDefault();
        const cityName = e.currentTarget.elements.cityName.value;
        e.currentTarget.elements.cityName.value = "";
        this.props.addFavorite(cityName);
        console.log(this.props.favorites.entries());

    }
}


function mapStateToProps(state) {
    return {
        favorites: state.fav.favorites,
        error: state.fav.error
    };
}

function mapDispatchToProps(dispatch) {
    return {
        addFavorite: (cityName) => {
            dispatch(addFavorite(cityName));
        },

        deleteFavorite: (cityName) => {
            dispatch(deleteFavorite(cityName));
        },

        loadWeatherWithName: (cityName) => {
            dispatch(loadWeatherWithName(cityName));
        },
        loadFavourites: () => {
            dispatch(loadFavourites())
        }
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Favorites);