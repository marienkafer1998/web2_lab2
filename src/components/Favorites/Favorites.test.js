import React from 'react';
import renderer from 'react-test-renderer';
import Favorites from "./Favorites.js";
import {Provider} from "react-redux";
import store from "../../store";

it('Favorites functional rendered properly', () => {
    const tree = renderer.create(
        <Provider store={store}>
            <Favorites/>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

