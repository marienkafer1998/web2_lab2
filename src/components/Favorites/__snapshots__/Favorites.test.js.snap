// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites functional rendered properly 1`] = `
<div
  className="favorite-weather container"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <h1
          className="fav"
        >
          Favorites
        </h1>
      </div>
      <div
        className="col-sm-5 offset-sm-1"
      >
        <form
          onSubmit={[Function]}
        >
          <input
            className="adder"
            name="cityName"
            placeholder="Add new city"
            required={true}
            text-transform="capitalize"
            type="text"
          />
          <button
            aria-label={null}
            className="btn btn-secondary"
            onClick={[Function]}
            type="submit"
          >
            +
          </button>
        </form>
      </div>
    </div>
  </div>
  <div
    className="row"
  />
</div>
`;

exports[`WeatherParam functional rendered properly 1`] = `
<div
  className="param"
>
  <div
    className="name"
  />
  <div
    className="value"
  />
</div>
`;

exports[`WeatherParameter functional rendered properly 1`] = `
<div
  className="param"
>
  <div
    className="name"
  />
  <div
    className="value"
  />
</div>
`;
